File: cpp/605.can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        // plant as left as possible
        int count = 0;
        for (int i = 0; i < flowerbed.size(); i++) {
            int left = max(i - 1, 0);
            int right = min(i + 1, ((int) flowerbed.size() - 1));
            if (flowerbed[i] == 0 && flowerbed[left] == 0 && flowerbed[right] == 0) {
                flowerbed[i] = 1;
                count++;
            }
        }
        return count >= n;
    }
};

File: python3/605.can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if n == 0:
            return True
        
        count = 0
        flowerbed = [0] + flowerbed + [0]
        for i in range(1, len(flowerbed) - 1):
            if not any(flowerbed[i - 1: i + 2]):
                count += 1
                flowerbed[i] = 1
                
                if count == n:
                    return True
                
        return False
            
        
